src/js/wall/__tests__/LoginForm.test.tsx

```diff
@@ -0,0 +1,74 @@
+import { screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import React from "react";
+import { useDispatch } from "react-redux";
+import { beforeEach, describe, expect, it, vi } from "vitest";
+import { renderWithProviders } from "../../../tests/setupTests";
+import { LoginForm } from "../LoginForm";
+import { useLoginMutation } from "../Queries";
+
+vi.mock("../Queries", () => ({
+    useLoginMutation: vi.fn(),
+}));
+
+vi.mock("react-redux", () => ({
+    useDispatch: vi.fn(),
+}));
+
+describe("<LoginForm />", () => {
+    let mockLoginMutation;
+    let mockDispatch;
+
+    beforeEach(() => {
+        mockDispatch = vi.fn();
+        (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
+
+        mockLoginMutation = {
+            mutate: vi.fn(),
+            isError: false,
+        };
+        (useLoginMutation as jest.Mock).mockReturnValue(mockLoginMutation);
```

Comment:
While this works okay, the preference is to mock via `nock`. nock intercepts API requests, allowing for more of the request lifecycle to be simulated and therefore tests the queries themselves as well.